### PR TITLE
feat: add price and price list in item supplier

### PR DESCRIPTION
**Pull request by hrwX**
- Add item price specified in price list for a supplier

Peek 2020-10-28 11-54

### CODE REVIEW

File: erpnext/stock/doctype/item/item.py [hunk 1]
```diff
@@ -1140,3 +1140,11 @@ def make_purchase_order_item(source_name, target_doc=None):
 	if item.item_defaults and item.item_defaults[0].default_supplier:
 		doc.supplier = item.item_defaults[0].default_supplier
 	return doc
+
+@frappe.whitelist()
+def get_item_price(supplier, item_code):
+	return frappe.db.get_value("Item Price", {
+		"supplier": supplier,
+		"item_code": item_code,
+		"buying": True
```
Comment (vishdha) [inline erpnext/stock/doctype/item/item.py:1149]: @hrwX  we have item reference too. so its better to add item details in items child table in purchase order.

@neilLasrado this button is in child table of supplier details which add 2 step to create PO. can we bring this button to Item master toolbar let user do multiple PO from dialog box if it has more supplier. @neilLasrado  what do you think?

File: erpnext/stock/doctype/item/item.js [hunk 2]
```diff
@@ -785,3 +785,31 @@ frappe.ui.form.on("UOM Conversion Detail", {
...
+				}
+			});
+		}
+	},
+	create_purchase_order: function(frm, cdt, cdn) {
+		let row = locals[cdt][cdn];
+		frappe.new_doc("Purchase Order", {
+			supplier: row.supplier,
+			buying_price_list: row.price_list
+		}, true);
```
Comment (vishdha) [inline erpnext/stock/doctype/item/item.js]: @hrwX  we have item reference too. so its better to add item details in items child table in purchase order.

@neilLasrado this button is in child table of supplier details which add 2 step to create PO. can we bring this button to Item master toolbar let user do multiple PO from dialog box if it has more supplier. @neilLasrado  what do you think?

Comment (hrwX) [reply erpnext/stock/doctype/item/item.js]: > @hrwX we have item reference too. so its better to add item details in items child table in purchase order.
- This is handled by erpnext, which will add the item details in the child table, so no need to do it maually.

Comment (vishdha) [reply erpnext/stock/doctype/item/item.js]: @hrwX I test this locally. not able to fetch item details in PO child table.
can you test it again

Comment (hrwX) [reply erpnext/stock/doctype/item/item.js]: @vishdha Have made a few changes and it should work now, please check

**Comment by vishdha** [comment]
@hrwX It's great that you have move functionality to the parent level. but still, we can give more control to user. I have modified the code. attached gif. let incorporate to item.

@neilLasrado what do you think about this. we can do for all buying cycles too.

Peek 2020-10-28 16-56

**Review by vishdha — COMMENTED** [review]
@hrwX It's great that you have move functionality to the parent level. but still, we can give more control to user. I have modified the code. attached gif. let incorporate to item.

@neilLasrado what do you think about this. we can do for all buying cycles too.

Peek 2020-10-28 16-56

**Comment by hrwX** [comment]
> @hrwX It's great that you have move functionality to the parent level. but still, we can give more control to user. I have modified the code. attached gif. let incorporate to item.
> 
> @neilLasrado what do you think about this. we can do for all buying cycles too.
> 
> Peek 2020-10-28 16-56

This functionality looks neat! But the requirements were to make one Purchase Order not multiple

**Comment by vishdha** [comment]
> > @hrwX It's great that you have move functionality to the parent level. but still, we can give more control to user. I have modified the code. attached gif. let incorporate to item.
> > @neilLasrado what do you think about this. we can do for all buying cycles too.
> > Peek 2020-10-28 16-56
> 
> This functionality looks neat! But the requirements were to make one Purchase Order not multiple

As UX wise if we have multiple suppliers then it should available for dialog let the user decide he/she want to create PO for a particular supplier. we have the functionality to remove supplier form dialog like this
Peek 2020-10-28 21-41

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**